inlong-manager/manager-service/src/main/java/org/apache/inlong/manager/service/datatype/KvDataTypeOperator.java

```diff
@@ -0,0 +1,97 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.inlong.manager.service.datatype;
+
+import org.apache.inlong.common.enums.DataTypeEnum;
+import org.apache.inlong.common.pojo.sort.dataflow.dataType.DataTypeConfig;
+import org.apache.inlong.common.pojo.sort.dataflow.dataType.KvConfig;
+import org.apache.inlong.manager.common.util.CommonBeanUtils;
+import org.apache.inlong.manager.pojo.consume.BriefMQMessage.FieldInfo;
+import org.apache.inlong.manager.pojo.stream.InlongStreamInfo;
+
+import lombok.extern.slf4j.Slf4j;
+import org.apache.commons.lang3.StringUtils;
+import org.springframework.stereotype.Service;
+
+import java.util.List;
+
+@Slf4j
+@Service
+public class KvDataTypeOperator implements DataTypeOperator {
+
+    @Override
+    public boolean accept(DataTypeEnum type) {
+        return DataTypeEnum.KV.equals(type);
+    }
+
+    @Override
+    public List<FieldInfo> parseFields(String str, InlongStreamInfo streamInfo) throws Exception {
+        List<FieldInfo> fields = CommonBeanUtils.copyListProperties(streamInfo.getFieldList(), FieldInfo::new);
+        try {
+            char separator = 0;
+            if (StringUtils.isNotBlank(streamInfo.getDataSeparator())) {
+                separator = (char) Integer.parseInt(streamInfo.getDataSeparator());
+            }
+            String[] bodys = StringUtils.split(str, separator);
+            if (bodys.length != fields.size()) {
+                return fields;
+            }
+            for (int i = 0; i < bodys.length; i++) {
+                String body = bodys[i];
+                char kvSeparator = '=';
+                if (StringUtils.isNotBlank(streamInfo.getKvSeparator())) {
+                    kvSeparator = (char) Integer.parseInt(streamInfo.getKvSeparator());
+                }
+                String[] values = StringUtils.split(body, kvSeparator);
+                fields.get(i).setFieldName(values[0]);
+                fields.get(i).setFieldValue(values[1]);
+            }
+        } catch (Exception e) {
+            log.warn("parse fields failed for groupId = {}, streamId = {}", streamInfo.getInlongGroupId(),
+                    streamInfo.getInlongStreamId(), e);
+        }
+        return fields;
+    }
+
+    @Override
+    public DataTypeConfig getDataTypeConfig(InlongStreamInfo streamInfo) {
+        char separator = 0;
+        if (StringUtils.isNotBlank(streamInfo.getDataSeparator())) {
+            separator = (char) Integer.parseInt(streamInfo.getDataSeparator());
+        }
+        Character escape = null;
+        if (streamInfo.getDataEscapeChar() != null) {
+            escape = streamInfo.getDataEscapeChar().charAt(0);
+        }
+        KvConfig kvConfig = new KvConfig();
+        char kvSeparator = '=';
+        Character lineSeparator = null;
```

Comment:
why `lineSeparator` doesn't have a default value.

Reply:
Because there are default values on the sort side.